### PR TITLE
Update usage examples to work with a DaemonApp

### LINKED ISSUE #5: Update usage examples to work with a DaemonApp base
Like https://github.com/sociomantic-tsunami/dhtproto/blob/neo/src/dhtproto/client/UsageExamples.d.

This makes the examples simpler for end-users to follow.

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**
This makes the examples simpler for end-users to follow, as they
would reflect the basic infrastructure that would be setup by the users.

Fixes #5

### CODE REVIEW

File: src/dlsproto/client/UsageExamples.d [hunk 1]
```diff
@@ -22,55 +22,77 @@ version (UnitTest)
...
 version (UnitTest)
 {
-    /// Helper struct containing a DLS client, an epoll instance, and basic
-    /// initialisation/connection code (This struct serves to avoid repeating
-    /// all the initialisation/connection boilerplate in every example. It is
-    /// not intended that real applications need such a struct. Typically,
-    /// its members would be in your main app class.)
-    struct DlsInit
+    // DaemonApp class showing typical neo DLS client initiialisation. The
+    // class has a single abstrac method -- example() -- which is implemented
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/UsageExamples.d]: abstrac

File: src/dlsproto/client/UsageExamples.d [hunk 2]
```diff
@@ -22,55 +22,77 @@ version (UnitTest)
...
+        }
+
+        // Application run method. Initializes the DLS client and starts the
+        // main application task.
+        override protected int run ( Arguments args, ConfigParser config )
+        {
             // Create a DLS client instance, passing the additional
-            // arguments required by neo: the authorisation name and
-            // password and the connection notifier (see below).
+            // arguemnts required by neo: the authorisation name and
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/UsageExamples.d]: arguemnts

File: src/dlsproto/client/UsageExamples.d [hunk 3]
```diff
@@ -22,55 +22,77 @@ version (UnitTest)
...
+                this.outer.example();
+            }
         }
 
+        // Abstract mehtod containing the logic for each example
+        abstract protected void example ( );
 
         // Notifier which is called when a connection establishment attempt
-        // succeeds or fails. (Also called after a re-connection attempt.)
+        // succeeds or fails. (Also called after a re-connection attempt)
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/UsageExamples.d]: Unrelated?

File: src/dlsproto/client/UsageExamples.d [hunk 4]
```diff
@@ -22,55 +22,77 @@ version (UnitTest)
...
-
-            this.connect(theScheduler.epoll);
+            // Thask entry point. Runs the example.
+            protected override void run ( )
+            {
+                this.outer.example();
+            }
         }
 
+        // Abstract mehtod containing the logic for each example
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/UsageExamples.d]: mehtod

File: src/dlsproto/client/UsageExamples.d [hunk 5]
```diff
@@ -86,32 +108,33 @@ version (UnitTest)
...
-
             // Assign a neo Put request. Note that the channel and value
             // are copied inside the client -- the user does not need to
-            // maintain them after calling this method.
-            this.dls_init.dls.neo.put("channel", 0x12345678, "value_to_put",
-                &this.putNotifier);
-
-            // Start the event loop to set it all running.
-            this.dls_init.epoll.eventLoop();
+            // mainain them after calling this method.
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/UsageExamples.d]: mainain

File: src/dlsproto/client/UsageExamples.d [hunk 6]
```diff
@@ -295,39 +309,18 @@ unittest
...
-        // Method which initialises the client and starts a request
-        public void start ( )
+        protected override void example ( )
         {
-            dls_init.connectWithScheduler();
-
-            // Start a task
-            theScheduler.schedule(new PutTask(this.dls_init.dls));
-            theScheduler.eventLoop();
+            // Perform a blocking neo Put request and return a result struct
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/UsageExamples.d:316]: Would be nice to have an example of the task-blocking Put with a proper notifier. (The one without the notifier isn't really recommended for use in applications.)

Comment (gavin-norman-sociomantic) [reply src/dlsproto/client/UsageExamples.d:316]: (dhtproto has both, with a note about when you might want to use the simple one: https://github.com/sociomantic-tsunami/dhtproto/blob/neo/src/dhtproto/client/UsageExamples.d#L226-L292)

**Review by gavin-norman-sociomantic — CHANGES_REQUESTED** [review]

File: src/dlsproto/client/UsageExamples.d (hunk 6 above)
Comment (nemanja-boric-sociomantic) [reply src/dlsproto/client/UsageExamples.d:316]: Yes, I am going to provide more usage examples, but I wanted to do after this PR.

Comment (gavin-norman-sociomantic) [reply src/dlsproto/client/UsageExamples.d:316]: Ok cool.

**Comment by nemanja-boric-sociomantic** [comment]
Updated.

**Review by gavin-norman-sociomantic — APPROVED** [review]
